clean up parameter spec/ref text

tnguyen-rh opened the pull request:
- describe ‘osc process -v’ arg using replaceables
- simplify description of valid parameter reference context
- add eol backslash for multiline command lines

@mfojtik WDYT?

mfojtik commented:
LGTM

@adellape PTAL

mfojtik commented:
(omg we should stop using acronyms...)

adellape commented on dev_guide/templates.adoc:
```diff
@@ -98,10 +98,9 @@ $ `osc process -f _<filename.json>_ | osc create -f -`
-template items.
+defined in the JSON file by adding the `-v` option followed by a
+comma-separated list of the form _<name>_=_<value>_.
+A parameter reference may appear in any text field inside the template items.
```
Just to be sure, we OK with dropping the specific "not restricted to only environment variables" bit? It was added in by @mfojtik's https://github.com/openshift/openshift-docs/pull/362 recently. The last sentence here does seem to still get that point across, just maybe not as overtly.

tnguyen-rh replied:
If env-var only was the situation w/ OpenShift v2, then i think it would be good to add back the emphasis that such a restriction is no longer the case w/ v3.  If env-var only was instead just a temporary implementation detail or design cul-de-sac, then i don't think it warrants mentioning.
@mfojtik @adellape 
So, which was it?

adellape commented on dev_guide/templates.adoc:
```diff
@@ -110,8 +109,8 @@ parameters to create a configuration with customized environment variables:
-$ osc process -f examples/sample-app/application-template-dockerbuild.json -v
-ADMIN_USERNAME=root,MYSQL_DATABASE=admin
+$ osc process -f examples/sample-app/application-template-dockerbuild.json \
+    -v ADMIN_USERNAME=root,MYSQL_DATABASE=admin
```
Instead of backslashes for multilines, I think we're preferring keeping long commands like this on a single line (an exception to the normal 80 column lines) and using `nowrap` to spawn a horizontal scroll. E.g.,:

```
====

[options="nowrap"]
----
$ osc process -f examples/sample-app/application-template-dockerbuild.json -v ADMIN_USERNAME=root,MYSQL_DATABASE=admin
----
====
```

Live example:

http://docs.openshift.org/latest/admin_guide/master_node_configuration.html#creating-new-configuration-files

adellape commented on dev_guide/templates.adoc:
```diff
@@ -122,8 +121,9 @@ uploading the template by piping the process output to the `create` command:
-ADMIN_USERNAME=root,MYSQL_DATABASE=admin | osc create -f -
+$ osc process -f examples/sample-app/application-template-dockerbuild.json \
+    -v ADMIN_USERNAME=root,MYSQL_DATABASE=admin \
+    | osc create -f -
```
Same comment re: `nowrap`.

tnguyen-rh replied:
@adellape 
So horizontal scroll is preferable?  Personally, i *really* dislike having to do it when reading documentation.  I think if backslashes can preserve correctness on cut and paste (as they do for shell commands), then why not use them?
Is there a guideline doc somewhere?  I'm sure this is not the first time the issue has surfaced...

adellape replied:
I don't disagree. It's come up previously, but not captured in any guidelines. A question for Bilhar.

tnguyen-rh replied:
@CowboysFan Are there any guidelines on horizontal scrolling use?  (Preferably, along the lines of: "Don't do it -- readers will hate you!"  :smile: )

CowboysFan commented:
@tnguyen-rh @adellape Our guidelines state to use the code block for all commands, which I believe automatically uses the "no-wrap" and the rendered output automatically uses the horizontal scroll bars where applicable. We did have an issue before with the sidebar block, which I fixed in the CSS to behave just as the code block, with the horizontal scroll bars. This is pretty widely used in the industry, and is the better option for web-focused documentation. As an example you can look at Rackspace docs, Heroku docs, and even the AppInfra docs within RH.

At this point we need to adhere to our own guidelines which have already been established, regardless of personal preference. These will be dictated by what our customers want, and I will collect that feedback frequently to make sure we're not confusing our customers. However, I'd prefer that we stick to our guidelines, and if anything is unclear, or if anything needs to be changed, then the process is to have a discussion before moving away from the established guidelines. 

We don't explicitly say it because if you just use the method in those guidelines where you mark it up with the code block, the rest takes care of itself; you will get the horizontal scroll bars. If you disagree with that, then rather than mark it up differently, you probably want to raise it as a discussion via email.

adellape commented:
@CowboysFan I'm fairly sure our default, `nowrap`-less codeblocks do not currently use horizontal scrollbars automatically. For example, compare the first `osadm` command here (that doesn't use `nowrap`) with others on the page (that do):

http://docs.openshift.org/latest/admin_guide/docker_registry.html

adellape commented:
@CowboysFan Actually, I was checking in Chrome only. It turns out scrollbars are rendered automatically for plain codeblocks when viewing in Firefox, but not Chrome. Dunno about other browsers. `nowrap` would at least make sure they appear no matter what.

tnguyen-rh commented:
I've added nowrap and fixed things up a little more.  I haven't modified the backslashes in the long command-lines, in light of section 3.4.3.1. Documenting Multiple or Long Commands, in the [Writing Style Guide](https://obriend.fedorapeople.org/WritingStyleGuide/).
@adellape PTAL

adellape commented:
@tnguyen-rh LGTM, can you squash?

tnguyen-rh commented:
@adellape Squashed.
